src/devices/BoardLed/BoardLed.cs

```diff
@@ -88,23 +112,25 @@ private void SetBrightness(int value)
                 value = 0;
             }
 
-            using StreamWriter writer = new StreamWriter(File.Open($"{DefaultDevicePath}/{Name}/brightness", FileMode.Open));
-
-            writer.Write(value);
+            _brightnessWriter.Write(value);
+            _brightnessWriter.Flush();
         }
 
-        private string GetTrigger()
-        {
-            using StreamReader reader = new StreamReader(File.Open($"{DefaultDevicePath}/{Name}/trigger", FileMode.Open, FileAccess.Read));
-
-            return Regex.Match(reader.ReadToEnd(), @"(?<=\[)(.*)(?=\])").Value;
-        }
+        private string GetTrigger() => Regex.Match(_triggerReader.ReadToEnd(), @"(?<=\[)(.*)(?=\])").Value;
 
         private void SetTrigger(string name)
         {
-            using StreamWriter writer = new StreamWriter(File.Open($"{DefaultDevicePath}/{Name}/trigger", FileMode.Open));
+            _triggerWriter.Write(name);
+            _triggerWriter.Flush();
+        }
 
-            writer.Write(name);
+        private void Initialize()
+        {
+            _brightnessReader = new StreamReader(File.Open($"{DefaultDevicePath}/{Name}/brightness", FileMode.Open, FileAccess.Read, FileShare.ReadWrite));
+            _brightnessWriter = new StreamWriter(File.Open($"{DefaultDevicePath}/{Name}/brightness", FileMode.Open, FileAccess.Write, FileShare.ReadWrite));
```

Comment:
you can actually re-use the same handle here but `leaveOpen: true` on half of them to not double dispose